### PR TITLE
languages/gleam: init

### PR DESCRIPTION
**Pull request by Soliprem**
Add TR and LSP support for gleam 

## Sanity Checking

Done it all 

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: modules/plugins/languages/gleam.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  options.vim.languages.gleam = {
+    enable = mkEnableOption "Gleam language support";
+
+    treesitter = {
+      enable = mkEnableOption "Gleam treesitter" // {default = config.vim.languages.enableTreesitter;};
+      package = mkOption {
+        type = package;
+        default = pkgs.vimPlugins.nvim-treesitter.builtGrammars.gleam;
+        description = "Gleam treesitter grammar to use";
+      };
```
Comment (NotAShelf) [inline modules/plugins/languages/gleam.nix:43]: Any reason we're not using `mkGrammarOption` here?

File: modules/plugins/languages/gleam.nix [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  inherit (lib.nvim.lua) expToLua;
+
+  cfg = config.vim.languages.gleam;
+
+  defaultServer = "gleam";
+  servers = {
+    gleam = {
+      package = pkgs.gleam;
+      lspConfig = ''
+        lspconfig.basedpyright.setup{
```
Comment (NotAShelf) [inline modules/plugins/languages/gleam.nix:21]: Wrong lspconfig template.

Comment (Soliprem) [reply modules/plugins/languages/gleam.nix:21]: damn me dumb

File: modules/plugins/languages/gleam.nix [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      server = mkOption {
+        type = enum (attrNames servers);
+        default = defaultServer;
+        description = "Gleam LSP server to use";
+      };
+
+      package = mkOption {
+        type = either package (listOf str);
+        default = servers.${cfg.lsp.server}.package;
+        description = "gleam LSP server package, or the command to run as a list of strings";
```
Comment (NotAShelf) [inline modules/plugins/languages/gleam.nix:55]:
```suggestion
        description = "Gleam LSP server package, or the command to run as a list of strings";
```

It's capitalized everywhere else, but not here.

**Review by NotAShelf — APPROVED** [review]

**Pull request merged**